File: server/src/main/java/com/fluffy/comment/domain/dto/AuthorDto.java
package com.fluffy.comment.domain.dto;

import com.querydsl.core.annotations.QueryProjection;

public record AuthorDto(
        Long id,
        String name,
        String avatarUrl
) {

    public static final AuthorDto EMPTY = new AuthorDto(0L, "", "");

    @QueryProjection
    public AuthorDto {
    }
}

File: server/src/main/kotlin/com/fluffy/comment/domain/dto/AuthorDto.kt
package com.fluffy.comment.domain.dto

data class AuthorDto(
    val id: Long,
    val name: String,
    val avatarUrl: String,
) {

    fun asDeleted() = AuthorDto(
        id = 0,
        name = "",
        avatarUrl = "",
    )
}
